### PR TITLE
docstring updates 

### PR DESCRIPTION
**Pull request by sosey**
see #2681  This includes updates, though not entirely complete, for assign_wcs, transforms, datamodels, lib, and extract_2d

### CODE REVIEW

File: jwst/lib/catalog_utils.py [hunk 1]
```diff
@@ -88,13 +87,37 @@ class SkyObject(namedtuple('SkyObject', ("sid",
...
+    ----------
+    sid : int
+        source identifed
+    xcentroid : float
+        x center of object in pixels
+    ycentroid : float
+        y center of object in pixels
+    sky_centroid: `~astropy.coordinates.SkyCorrd`
+        ra and dec of the center of the object
+    abmag : float64
```
Comment (nden) [inline jwst/lib/catalog_utils.py]: `float`

File: jwst/lib/catalog_utils.py [hunk 2]
```diff
@@ -88,13 +87,37 @@ class SkyObject(namedtuple('SkyObject', ("sid",
...
+        AB Magnitude of object
+    abmag_error : float32
```
Comment (nden) [inline jwst/lib/catalog_utils.py]: `float`

File: jwst/lib/catalog_utils.py [hunk 3]
```diff
@@ -88,13 +87,37 @@ class SkyObject(namedtuple('SkyObject', ("sid",
...
+        Error on the AB magnitude
+    sky_bbox_ll : `~astropy.coordinates.SkyCorrd`
```
Comment (nden) [inline jwst/lib/catalog_utils.py]: typo `SkyCoord` (and further down)

File: jwst/transforms/models.py [hunk 4]
```diff
@@ -57,6 +57,31 @@ class GrismObject(namedtuple('GrismObject', ("sid",
...
+    Parameters
+    ----------
+    sid : int
+        source identifed
+    xcentroid : float
+        x center of object in pixels
+    ycentroid : float
+        y center of object in pixels
+    order_bounding : dict
+        Contains the object x,y bounding location on the image
```
Comment (nden) [inline jwst/transforms/models.py]: Is this a mapping `{order: (x,y)}`?

File: jwst/transforms/models.py [hunk 5]
```diff
@@ -57,6 +57,31 @@ class GrismObject(namedtuple('GrismObject', ("sid",
...
+    partial_order : bool
+        True if the order is only partially contained on the image
+    waverange : list
+        wavelength range for the order
+    sky_centroid: `~astropy.coordinates.SkyCorrd`
```
Comment (nden) [inline jwst/transforms/models.py]: SkyCoord typo

File: jwst/assign_wcs/assign_wcs.py [hunk 6]
```diff
@@ -97,6 +97,13 @@ def _add_3rd_axis(input_model):
     """
     Add WCS keywords and their default values for the spectral axis.
 
+    Parameters
+    ----------
+    input_model: `~jwst.datamodels.DataModel`
```
Comment (philhodge) [inline jwst/assign_wcs/assign_wcs.py]: The numpy documentation for docstrings says that in parameter descriptions, a space is required before the colon.  I saw a comment on reddit that says the documentation won't be built correctly if the space is missing.  Is this true?  Do we really need a space both before and after the colon?  If so, there are a number of cases where the space is missing.

File: jwst/assign_wcs/nircam.py [hunk 7]
```diff
@@ -204,20 +250,26 @@ def wfss(input_model, reference_files):
...
         The input datamodel, derived from datamodels
     reference_files: dict
         Dictionary {reftype: reference file name}.
 
+    Returns
+    -------
+    pipeline : list
+        The pipeline list that is returned is suitable for
+        input into  gwcs.wcs.WCS to create a GWCS object.
+
```
Comment (philhodge) [inline jwst/assign_wcs/nircam.py]: In the Notes section just below, is the first line a complete sentence?  It looks as if the first line should end with a period and the second line should begin with "Not".

File: jwst/assign_wcs/nircam.py [hunk 8]
```diff
@@ -204,20 +250,26 @@ def wfss(input_model, reference_files):
...
     -----
     The tree in the grism reference file has a section for each order/beam
     not sure if there will be a separate passband reference file needed for
     the wavelength scaling or wedge offsets. This helper is currently in
     jwreftools/nircam/nircam_reftools.
 
-    The direct image the catalog has been created from was corrected for
-    distortion, but the dispersed images have not. This is OK if the trace and
+    The direct image the catalog has been created from was processed through
+    resampe, but the dispersed images have not. This is OK if the trace and
```
Comment (philhodge) [inline jwst/assign_wcs/nircam.py]: Typo:  resampe --> resample

File: jwst/transforms/models.py (hunk 4 above)
Comment (sosey) [reply jwst/transforms/models.py]: It's a member of the GrismObject, a dictionary keyed on spectral order that contains a tuple of ((ymin, ymax), (xmin, xmax))

File: jwst/extract_2d/grisms.py [hunk 9]
```diff
@@ -158,7 +159,7 @@ def extract_tso_object(input_model,
 
         if (extract_y_min > extract_y_max):
             raise ValueError("Something bad happened calculating extraction y-size")
-          
+
```
Comment (philhodge) [inline jwst/extract_2d/grisms.py]: Typo in the next line:  "boxto" --> "box to"

File: jwst/extract_2d/grisms.py [hunk 10]
```diff
@@ -244,6 +246,36 @@ def extract_grism_objects(input_model,
...
+    Step 2: Record source information for each object in the catalog: position
+            (RA and Dec), shape (A_IMAGE, B_IMAGE, THETA_IMAGE), and all
+            available magnitudes, and minimum bounding boxes
+
+    Step 3: Compute the trace and wavelength solutions for each object in the
+            catalog and for each spectral order.  Record this information.
+
+    Step 4: Compute the WIDTH of each spectral subwindow, which may be fixed or
+            variable. The cross-dispersion size is taken from the minum bounding
```
Comment (philhodge) [inline jwst/extract_2d/grisms.py]: Typo:  minum --> minimum

File: jwst/assign_wcs/assign_wcs.py (hunk 6 above)
Comment (sosey) [reply jwst/assign_wcs/assign_wcs.py]: I would think it only made a spacing difference, but we might as well be consistent.

Comment (nden) [reply jwst/assign_wcs/assign_wcs.py]: I think there needs to be a space. I have not tested it withut but have been told strictly that there should be a space.

**Pull request merged**